src/react.tsx

```diff
@@ -15,23 +38,36 @@ export function connect<StoreState extends object>(store: StoreDefinition<StoreS
   ): React.ComponentClass<Diff<PropsWithStore, { store: Store<StoreState> }>> {
 
     type State = {
-      store: StoreSnapshot<StoreState>
-      subscription: Subscription
+      store: StoreSnapshotWrapper<StoreState>
     }
 
     return class extends React.Component<Diff<PropsWithStore, { store: Store<StoreState> }>, State> {
       static displayName = `withStore(${getDisplayName(Component)})`
+      const subscriptions: Map<keyof StoreState, Subscription> = new Map()
```

Comment:
This should go on `state`, as we already prevent extra re-renders with `shouldComponentUpdate`.

Also, no need to manually type it. In TS (as in Flow starting a few weeks ago), you can bind polymorphic type parameters explicitly.

```diff
-const subscriptions: Map<keyof StoreState, Subscription> = new Map()
+const subscriptions = new Map<keyof StoreState, Subscription>()
```

Reply:
Why should it go in the state? In my experience, subscriptions are never part of the state as the rendering never depends on them and they are detached from the normal React workflow.

Reply:
I try to keep all instance data on `state`, so it's easy to tell what's code (common to all instances) vs. data (specific to each instance).

Reply:
That forces you to write shouldComponentUpdate functions, this way is easier I think, but not going to argue on this, its your project